erts/emulator/nifs/common/prim_socket_nif.c

```diff
@@ -7001,6 +7021,12 @@ ERL_NIF_TERM esock_accept_accepting_current_accept(ErlNifEnv*       env,
     if (esock_accept_accepted(env, descP, sockRef, accSock,
                               descP->currentAcceptor.pid, &res)) {
 
+        ESOCK_ASSERT( DEMONP("esock_accept_accepting_current_accept -> "
+                             "current acceptor",
+                             env, descP, &descP->currentAcceptor.mon) == 0);
+
+        MON_INIT(&descP->currentAcceptor.mon);
```

Comment:
Is it necessary to `MON_INIT` after demonitoring is done?

Reply:
I have taken the test code and added it to socket_SUITE (with some minor modifications).
MON_INIT is mostly *really* sure...